queries.sql

```diff
@@ -8,4 +8,54 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon','Pikachu');
 SELECT name,escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE name!='Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+
+BEGIN TRANSACTION;
+SAVEPOINT SP1;
+ROLLBACK TO SP1;
+ROLLBACK;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon%';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth BETWEEN '2022-1-1' AND '2050-1-1';
+SAVEPOINT SP1;
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO SP1;
+UPDATE animals
+SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(name) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT MAX(escape_attempts) FROM animals;
+SELECT COUNT(neutered) FROM animals WHERE neutered = false;
+SELECT COUNT(neutered) FROM animals WHERE neutered = true;
+
```

Comment:
- [ ] Same here. 🤔 You had to select Who escapes the most, neutered or not neutered animals? 😄 It should look like this:

```
   SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```

or

```
  SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```

- Please correct that 😉 👍🏻